src/test/java/htsjdk/samtools/util/DateParserTest.java

```diff
@@ -0,0 +1,173 @@
+  3. Notice of any changes or modifications to the W3C files, including the
+     date changes were made. (We recommend you provide URIs to the location
+     from which the code is derived).
+
+In addition, creators of derivitive works must include the full text of this
+NOTICE in a location viewable to users of the derivitive work.
+
+THIS SOFTWARE AND DOCUMENTATION IS PROVIDED "AS IS," AND COPYRIGHT HOLDERS
+MAKE NO REPRESENTATIONS OR WARRANTIES, EXPRESS OR IMPLIED, INCLUDING BUT NOT
+LIMITED TO, WARRANTIES OF MERCHANTABILITY OR FITNESS FOR ANY PARTICULAR
+PURPOSE OR THAT THE USE OF THE SOFTWARE OR DOCUMENTATION WILL NOT INFRINGE
+ANY THIRD PARTY PATENTS, COPYRIGHTS, TRADEMARKS OR OTHER RIGHTS.
+
+COPYRIGHT HOLDERS WILL NOT BE LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR
+CONSEQUENTIAL DAMAGES ARISING OUT OF ANY USE OF THE SOFTWARE OR
+DOCUMENTATION.
+
+The name and trademarks of copyright holders may NOT be used in advertising
+or publicity pertaining to the software without specific, written prior
+permission. Title to copyright in this software and any associated
+documentation will at all times remain with copyright holders.
+
+____________________________________
+
+This formulation of W3C's notice and license became active on August 14
+1998. See the older formulation for the policy prior to this date. Please
+see our Copyright FAQ for common questions about using materials from our
+site, including specific terms and conditions for packages like libwww,
+Amaya, and Jigsaw. Other questions about this notice can be directed to
+site-policy@w3.org .
+
+
+
+
+webmaster
+(last updated 14-Aug-1998)
+
+ */
+
+package htsjdk.samtools.util;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.text.DateFormat;
+import java.util.Date;
+
+/**
+ * NOTE: This code has been taken from w3.org, and modified slightly to handle timezones of the form [-+]DDDD,
+ * and also to fix a bug in the application of time zone to the parsed date.
+ *
+ * Date parser for ISO 8601 format
+ * http://www.w3.org/TR/1998/NOTE-datetime-19980827
+ * @version $Revision: 1.3 $
+ * @author  bmahe@w3.org
+ */
+
+public class DateParserTest {
+
+    private static void test(final String isodate) {
+        System.out.println("----------------------------------");
+        try {
+            Date date = DateParser.parse(isodate);
+            System.out.println(">> " + isodate);
+            System.out.println(">> " + date.toString() + " [" + date.getTime() + "]");
+            final String isodateRoundTrip = DateParser.getIsoDate(date);
+            System.out.println(">> " + isodateRoundTrip);
+
+            final Date orig = DateParser.parse(isodate);
+            final Date roundTrip = DateParser.parse(isodateRoundTrip);
+
+            assertDatesAreClose(orig, roundTrip);
+
+        } catch (DateParser.InvalidDateException ex) {
```

Comment:
I think we still need to remove these catch clauses because they could hide the fact that none of the dates actually parse.